File: app/src/main/java/com/chess/cryptobot/view/adapter/BalanceViewOnClickListener.java
package com.chess.cryptobot.view.adapter;

import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.chess.cryptobot.content.balance.BalanceHolder;
import com.chess.cryptobot.content.balance.BalancePreferences;
import com.chess.cryptobot.view.dialog.MinBalanceDialog;

public class BalanceViewOnClickListener implements RecyclerViewOnClickListener {
    private BalanceHolder balanceHolder;

    BalanceViewOnClickListener(BalanceHolder balanceHolder) {
        this.balanceHolder = balanceHolder;
    }

    @Override
    public void onClick(View view, String coinName) {
        BalancePreferences balancePreferences = (BalancePreferences) balanceHolder.getPrefs();
        Double minBalance = balancePreferences.getMinBalance(coinName);
        MinBalanceDialog dialog = new MinBalanceDialog(balanceHolder, coinName, minBalance);
        AppCompatActivity activity = (AppCompatActivity) balanceHolder.getContext();
        dialog.show(activity.getSupportFragmentManager(), "coinName");
    }
}

File: app/src/main/java/com/chess/cryptobot/view/adapter/BalanceViewOnClickListener.kt
package com.chess.cryptobot.view.adapter

import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.chess.cryptobot.content.balance.BalanceHolder
import com.chess.cryptobot.content.balance.BalancePreferences
import com.chess.cryptobot.view.dialog.MinBalanceDialog

class BalanceViewOnClickListener internal constructor(private val balanceHolder: BalanceHolder) : RecyclerViewOnClickListener {
    override fun onClick(view: View?, coinName: String?) {
        val balancePreferences = balanceHolder.prefs as BalancePreferences
        val minBalance = balancePreferences.getMinBalance(coinName!!)
        val dialog = MinBalanceDialog(balanceHolder, coinName, minBalance)
        val activity = balanceHolder.context as AppCompatActivity?
        dialog.show(activity!!.supportFragmentManager, "coinName")
    }

}